File: check-images.js
// check-images.js
import fs from 'fs';
import path from 'path';

// مسار ملف البيانات ومسار مجلد الصور
const dataPath = path.resolve('src/booksData_Complete.json');
const imagesFolder = path.resolve('public/books');

const rawData = fs.readFileSync(dataPath, 'utf-8');
const booksData = JSON.parse(rawData);

let missingCount = 0;
let totalBooks = 0;

console.log('🔍 جاري فحص صور الأغلفة...\n');
console.log('-----------------------------------------');

Object.entries(booksData).forEach(([stage, grades]) => {
  Object.entries(grades).forEach(([grade, books]) => {
    books.forEach((book) => {
      totalBooks++;
      // مسار الصورة المفترض
      const expectedImagePath = path.join(imagesFolder, `${book.id}.jpg`); 
      const expectedImagePathPng = path.join(imagesFolder, `${book.id}.png`);
      const expectedImageName = path.join(imagesFolder, `${book.name}.jpg`);

      // التحقق هل الصورة موجودة بأي من هذه الأسماء؟
      if (!fs.existsSync(expectedImagePath) && !fs.existsSync(expectedImagePathPng) && !fs.existsSync(expectedImageName)) {
        console.log(`❌ مفقودة: [${stage} - ${grade}] -> اسم الكتاب: ${book.name}`);
        missingCount++;
      }
    });
  });
});

console.log('-----------------------------------------');
console.log(`📊 النتيجة: من أصل ${totalBooks} كتاب، هناك ${missingCount} صورة مفقودة أو اسمها خاطئ.`);

File: check_images.py
import json
import os

data_path = 'src/booksData_Complete.json'
images_folder = 'public/books'

with open(data_path, 'r', encoding='utf-8') as f:
    data = json.load(f)

existing_images = os.listdir(images_folder)
print("🔍 فحص تطابق الصور مع البيانات للصف الثاني الثانوي:\n")

target_stage = 'الثانوية'
target_grade = 'الثاني الثانوي'

if target_stage in data and target_grade in data[target_stage]:
    for book in data[target_stage][target_grade]:
        # جلب الـ ID، وإذا لم يكن موجوداً نطبع تحذيراً
        book_id = book.get('id', 'بدون_ID')
        image_name = f"{book_id}.jpg"
        
        if image_name not in existing_images:
            print(f"❌ مفقودة: {book['name']} (يبحث عن صورة باسم: {image_name})")
        else:
            print(f"✅ موجودة: {book['name']}")
else:
    print("خطأ في قراءة بيانات الصف.")